### PR TITLE
Deployment Logs

### LINKED ISSUE #37: Access deployment logs
```
inertia [REMOTE] logs -c [MY_CONTAINER_NAME]
```

Allow users to view/stream logs from their inertia deployment

### PR DESCRIPTION
**Pull request by bobheadxi**
:hourglass: **Status**: Ready

:tickets: **Ticket(s)**: Closes #37

---

## :construction_worker: Changes

- Stream Inertia daemon logs with `inertia local logs --stream`, or access most recent logs with `inertia local logs`
- Stream docker-compose logs or specific containers using `inertia local logs docker-compose` or `inertia local logs mycontainer` specifically
-Reworked logging, ~~though this is bugged right now:~~

~~https://github.com/ubclaunchpad/inertia/blob/rob/%2337-deployment-logs/daemon/handler.go#L196 doesn't seem to properly return the error to the client - I think its still floating around somewhere because if I run `inertia local up` again, the line that was supposed to be returned shows up in the container logs alongside the first log from the second call~~ 
FIXED as of 754d625, turns out a `newline` is needed for the flush to trigger

## :flashlight: Testing Instructions

```bash
inertia local init
inertia local up
inertia local logs --stream # stream inertia daemon logs
inertia local logs docker-compose --stream
```

### CODE REVIEW

File: common/types.go [hunk 1]
```diff
@@ -22,8 +22,9 @@ const (
...
 
-// UpRequest is the body of a up request to the daemon.
-type UpRequest struct {
-	Stream bool   `json:"stream"`
-	Repo   string `json:"repo"`
+// DaemonRequest is the configurable body of a request to the daemon.
+type DaemonRequest struct {
+	Stream    bool   `json:"stream"`
+	Repo      string `json:"repo,omitempty"`
+	Container string `json:"container,omitempty"`
```
Comment (chadlagore) [inline common/types.go]: As handy as this JSON language is, it feels like a smell to me

File: daemon/handler.go [hunk 2]
```diff
@@ -48,6 +51,39 @@ const (
...
+		log.Println("Failed to pull docker-compose image - shutting down daemon.")
+		cli.Close()
+		return
+	}
+	cli.Close()
+
+	// Run daemon on port
+	log.Println("Serving daemon on port " + port)
+	mux := http.NewServeMux()
+	// Example usage of `authorized' decorator.
```
Comment (chadlagore) [inline daemon/handler.go]: Can kill my old silly comments, think it’s clear what’s goin down here now😜

File: common/types.go (hunk 1 above)
Comment (bobheadxi) [reply common/types.go]: hm what do you mean?

Comment (chadlagore) [reply common/types.go]: It just feels like it got hacked onto the language. Its very handy, but something feels strange about it. I'm just murmuring out loud, ignore me :+1:

Comment (bobheadxi) [reply common/types.go]: true, functionality-wise I think it's quite nice though 😛

**Pull request merged**